src/optimizer.jl

```diff
@@ -0,0 +1,179 @@
+# Supported Functions
+const SF = Union{MOI.SingleVariable, 
+                 MOI.ScalarAffineFunction{Float64}, 
+                 MOI.VectorOfVariables, 
+                 MOI.VectorAffineFunction{Float64}}
+
+# Supported Sets
+const SS = Union{MOI.EqualTo{Float64}, MOI.GreaterThan{Float64}, MOI.LessThan{Float64}, 
+                 MOI.Zeros, MOI.Nonnegatives, MOI.Nonpositives, 
+                 MOI.SecondOrderCone, MOI.RotatedSecondOrderCone,
+                 MOI.ExponentialCone, MOI.DualExponentialCone,
+                 MOI.PowerCone, MOI.DualPowerCone,
+                 MOI.PositiveSemidefiniteConeTriangle}
+
+mutable struct DualOptimizer <: MOI.AbstractOptimizer
+    dual_problem::Union{Nothing, Dualization.DualProblem}
+    dual_optimizer::Union{Nothing, MOI.AbstractOptimizer}
```

Comment:
I would do `dual_optimizer::OT` et `DualOptimizer{OT}`.
Otherwise, accessing this field is type instable and it will be slow

Reply:
yup, that makes sense!

Reply:
what does OT stands for?

Reply:
**O**ptimizer **T**ype

Reply:
I tried like this but it doesn't work
```
mutable struct DualOptimizer{OT <: MOI.ModelLike} <: MOI.AbstractOptimizer
    dual_problem::Union{Nothing, DualProblem}
    dual_optimizer::OT
    dual_optimizer_idx_map::Union{Nothing, MOIU.IndexMap}

    function DualOptimizer{OT}(dual_optimizer::OT) where {OT <: MOI.ModelLike}
        return new(nothing, dual_optimizer, nothing)
    end
end
```
It gives me and error
`ERROR: LoadError: LoadError: MethodError: no method matching Dualization.DualOptimizer(::GLPK.Optimizer)`

Reply:
I think it should be:
```julia
    function DualOptimizer{OT}(dual_optimizer::OT) where {OT <: MOI.ModelLike}
        return new(nothing, dual_optimizer, nothing)
    end
    function DualOptimizer(dual_optimizer::OT) where {OT <: MOI.ModelLike}
        return DualOptimizer{OT}(dual_optimizer)
    end
```

Reply:
true, just saw what I missed, the constructor should not have '{OT}'